indico/filters/__init__.py

```diff
@@ -59,33 +59,73 @@ def __init__(
         super().__init__(**kwargs)
 
 
+class DateRangeFilter(dict):
+    """
+    Create a Filter when querying for Submissions within a certain date range
+    Args:
+        filter_from (str): A valid string representation of a datetime for start date to filter
+        filter_to (str): A valid string representation of a datetime for end date to filter
+    """
+
+    def __init__(self, filter_from: str = None, filter_to: str = None):
```

Comment:
nit: typing should be `str | None` if we default to None. However maybe typing in the sdk is different, if so no worries

Reply:
typing should match the lowest supported version of python, so should be `Union[ str, None ]` to match 3.9+